client/tasks.py

```diff
@@ -45,7 +45,7 @@ def handle_response(response_details):
         In case of any internal server error, retry that task again.
         """
         log.warning('Retrying: {} task for {}: {}'.format(task_name, job_type, entity))
-        caller.retry()
+        caller.retry(exc=None, countdown=int(random.uniform(2, 4) ** caller.request.retries))
```

Comment:
I don't think it's an exponential wait. Please look into this. Can you please explain how it will work and generate exponential wait?

Reply:
done.